rms/search_engine/clients.py

```diff
@@ -71,3 +74,59 @@ def _build_abstract_keywords_criteria(abstract_keywords: list[str] | None) -> st
             return ""
         abstract_keywords_criteria = " OR ".join([f'ABS("{keyword}")' for keyword in abstract_keywords])
         return f"({abstract_keywords_criteria})"
+
+
+class DblpClient(ABC):
+
+    def __init__(self):
+        self.settings = Settings()
+        self.headers = {
+            "httpAccept": "application/json",
+        }
+        self.dblp_search_api_endpoint = self.settings.dblp_search_endpoint
+
+
+class DblpAuthorSearchApi(DblpClient):
+
+    def __init__(self):
+        super().__init__()
+
+    async def search(self, query: DblpAuthorSearchBody) -> dict[str, Any]:
+        params = self.build_params(query)
+
+        async with httpx.AsyncClient() as client:
+            response = await client.get(self.dblp_search_api_endpoint, headers=self.headers, params=params)
+            response.raise_for_status()
+
+            return response.json()
+
+    @staticmethod
+    def build_params(query: DblpAuthorSearchBody) -> dict[str, Any]:
+        params = {
+            "q": query.author_name,
+            "format": query.format
+        }
+
+        return params
+
+
+class ScholarClient(ABC):
+
+    def __init__(self):
+        self.sections = ["basics", "indices"]
+
+
+class ScholarAuthorSearchApi(ScholarClient):
+
+    def __init__(self):
```

Comment:
Not needed when initialization behavior is not changed. Init method from super class will be called automatically

Reply:
fixed